### PR TITLE
Fix inventory locks

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

**Comment by trasher** [comment]
Sorry for the late last commit, I should have seen that before.

File: src/Inventory/Asset/Peripheral.php [hunk 1]
```diff
@@ -163,10 +162,13 @@ public function handle()
...
                 if ($data['found_inventories'][0] == 0) {
                     // add peripheral
-                    $handled_input['entities_id'] = $this->entities_id;
+                    $handled_input = $this->handleInput($val, $peripheral) + ['entities_id' => $this->entities_id];
                     $items_id = $peripheral->add(Toolbox::addslashes_deep($handled_input), [], false);
                 } else {
                     $items_id = $data['found_inventories'][0];
+                    $peripheral->getFromDB($items_id);
+                    $handled_input = $this->handleInput($val, $peripheral);
+                    //FIXME: handleInput
```
Comment (cedric-anne) [inline src/Inventory/Asset/Peripheral.php:171]:
```suggestion
```

File: tests/functionnal/Lockedfield.php [hunk 2]
```diff
@@ -602,4 +602,309 @@ public function testLockedDdValue()
...
+        if ($inventory->inError()) {
+            $this->dump($inventory->getErrors());
+        }
+        $this->boolean($inventory->inError())->isFalse();
+        $this->array($inventory->getErrors())->isEmpty();
+
+        //check created databases & instances
+        $this->integer(countElementsInTable(\DatabaseInstance::getTable()))->isIdenticalTo(1);
+
+        //ensure database version has been updated
```
Comment (cedric-anne) [inline tests/functionnal/Lockedfield.php:903]:
```suggestion
        //make sure manufacturer is still the correct one
```

**Review by cedric-anne — APPROVED** [review]
I'm a bit lost after all these changes made on locks management, but it seems OK.

**Comment by trasher** [comment]
> I'm a bit lost after all these changes made on locks management, but it seems OK.

So am I :D We've added tests, we fixed all that together with @stonebuzz; and requester has confirmed this fix his issues.

**Review by stonebuzz — APPROVED** [review]

**Comment by stonebuzz** [comment]
Just a remark (this will be part of another PR)
a lock on the OS architecture is not displayed in the "lock" tab

image

image

**Comment by trasher** [comment]
> Just a remark (this will be part of another PR) a lock on the OS architecture is not displayed in the "lock" tab

Lock tab shows only locks for current itemtype... You will only see locks on current computer, never OS devices, etc. But indeed, that is anothher problem, so another PR ;)

**Pull request merged**